app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true

```

Comment:
- [ ] You are using a fully complete users controller, however, the app never asks for authentication (just the username is enough).  Please, kindly add the authentication functionality.